BUILDIT.md

```diff
@@ -1546,27 +1546,21 @@ defmodule AppWeb.AppLive do
     end
   end
 
-  # assign default values to socket:
-  defp assign_socket(socket) do
-    person_id = get_person_id(socket.assigns)
-    assign(socket, items: Item.items_with_timers(person_id), active: %Item{}, editing: nil)
-  end
-
   @impl true
   def mount(_params, _session, socket) do
     # subscribe to the channel
     if connected?(socket), do: AppWeb.Endpoint.subscribe(@topic)
-    {:ok, assign_socket(socket)}
+    # we load the items in the `handle_params` function which is called
+    # after mount is finished
+    {:ok, assign(socket, items: [], editing: nil, filter: "active")}
   end
 
   @impl true
   def handle_event("create", %{"text" => text}, socket) do
     person_id = get_person_id(socket.assigns)
     Item.create_item(%{text: text, person_id: person_id, status: 2})
-    # IO.inspect(socket.assigns, label: "handle_event create socket.assigns")
-    socket = assign_socket(socket)
 
-    AppWeb.Endpoint.broadcast_from(self(), @topic, "update", socket.assigns)
+    AppWeb.Endpoint.broadcast(@topic, "update", socket.assigns)
```

Comment:
Using `broadcast` instead of `broadcast_from` to also use the `update` event to update the UI of the client triggering the event. This allow us to retrieve the list of items only once (instead of twice)

Reply:
My understanding was that [`broadcast_from/4`](https://hexdocs.pm/phoenix/Phoenix.Endpoint.html#c:broadcast_from/4) would inform the handler not to waste time on the message for the process (`self()`) that sent the message i.e. send the message to everyone, but when I receive my own message, ignore it. 💭 
Whereas [`broadcast/3`](https://hexdocs.pm/phoenix/Phoenix.Endpoint.html#c:broadcast/3) would not have that information so we'd end up with redundant updates.